src/child/components/Favourite.js

```diff
@@ -18,18 +17,19 @@ class Favourite extends Component {
     componentDidMount() {
         const stockCode = this.props.stockCode;
         this.addDragTarget(stockCode);
-        quandlService.getStockData(stockCode, response => {
-            const data = response.stockData.data[0];
-            const stockData = {
-                name: response.dataset.name,
-                price: data.close,
-                delta: data.close - data.open,
-                percentage: (data.close - data.open) / data.open * 100
-            };
-            const chartData = response;
-            this.props.bindings.onQuandlResponse(stockCode, response.dataset.name);
-            this.setState({ stockData, chartData });
-        });
+        quandlServiceGetStockData(stockCode)
+            .then(response => {
+                const data = response.stockData.data[0];
+                const stockData = {
+                    name: response.dataset.name,
```

Comment:
Not sure if its worth it, but could pull this out:
```JavaScript
const stockName = response.dataset.name
```

It would just keep it consistent with how we're passing stockCodes and names around below as well

```JavaScript
this.props.bindings.onQuandlResponse(stockCode, stockName);
```